src/cdk/table/row.ts

```diff
@@ -294,3 +294,11 @@ export class CdkFooterRow {
 })
 export class CdkRow {
 }
+
+/** Row that can be used to display a message when no data is shown in the table. */
+@Directive({
+  selector: 'ng-template[cdkNoDataRow]'
```

Comment:
The name of the directive sounds a little clunky, but I couldn't come up with anything more descriptive. I'm open ideas on how to name it better.

Reply:
The term "placeholder" comes to (my) mind but I'm not sure it'd be adequate here.

I'd somewhat like to have all the elements that are "placeholder" or some sort identified as such across all the components that use them. Not all the placeholders in particular, but all the elements that are (semantically) somewhat agnostic to the "collection" they belong to in general.

A reusable naming convention makes for a conceptually simpler API!

I think the CDK Drag&Drop uses something like `cdkDragPlaceholder`, so maybe `cdkNoDataPlaceholder` would be acceptable... but I don't like it, it's clunkier than `cdkNoDataRow` ever was 🤪

Disclaimer: I really haven't thought this through, this is just my two cents.

Reply:
Having "placeholder" in there sounds good to me since this behaves a little bit like the native `input` placeholder (e.g. something that shows up when the input is empty).